### PR TITLE
Fhir 40032, fhir 46287

### PR DESCRIPTION
**Pull request by dand9959**
FHIR-40032 
    --- $care-gaps operation:
        Added subjectGroup, measureResource and measure IN parameters to $care-gaps operation
	    Changed $care-gaps "type" to "instance" so that the URL: [base]/Measure/[id]/$care-gaps is a valid option	
  ---	$deqm.evaluate-measure operation:
	    Renamed operation $deqm.evaluate-measure to $evaluate
	    Added subjectGroup parameter to $evaluate operation
	
 --- $deqm.collect-data operation
	    Added new operation:  $deqm.collect-data

FHIR-46287
	$evaluate operation:
		Added measureId, measureIdentifier, measureUrl, measureResource and report IN parameters
		Change cardinaltiy of measure IN parameter to 0.**

### CODE REVIEW

**Comment by dand9959** [comment]
[DEQM Operations Update.xlsx](https://github.com/user-attachments/files/16448244/DEQM.Operations.Update.xlsx)
This is the final spreadsheet from Matt Tiller outlining the change implemented (for FHIR-40032 and FHIR-46287 only)

**Review by yanheras — CHANGES_REQUESTED** [review]

**Comment by dand9959** [comment]
Regarding last commit:  [5de5cc1](https://github.com/HL7/davinci-deqm/pull/136/commits/5de5cc11d0011c827b989f561941f56ccb6495ff)

Updated links for manifest parameter target profile, and also for parameter programCode valueSet.  These links will result in QA errors at this point, but after dependencies are update, these will work.

**Comment by dand9959** [comment]
No documentation was provided for the aforementioned parameters.  Aside from the organization documentation suggested in comments here, further guidance/verbiage for the others can be easily added once provided.

**Review by yanheras — APPROVED** [review]

File: archived/resources/OperationDefinition-deqm.evaluate-measure.xml [hunk 1]
```diff
@@ -44,6 +44,13 @@
         <type value="date"/>
     </parameter>
     <parameter>
+      <name value="program"/>
+      <use value="in"/>
+      <min value="0"/>
+      <max value="1"/>
+      <documentation value="The quality program being reported via canonical reference to ([Library{CQFMQualityProgram}](http://hl7.org/fhir/us/cqfmeasures/StructureDefinition/library-cqfm)) indicating which quality program is being reported."/>
```
Comment (p9g) [inline archived/resources/OperationDefinition-deqm.evaluate-measure.xml:51]: https://jira.hl7.org/browse/FHIR-45309 renames CQFMQualityProgram to CQFMContentRelease, not yet applied

File: archived/resources/OperationDefinition-deqm.evaluate-measure.xml [hunk 2]
```diff
@@ -44,6 +44,13 @@
         <type value="date"/>
     </parameter>
     <parameter>
+      <name value="program"/>
```
Comment (p9g) [inline archived/resources/OperationDefinition-deqm.evaluate-measure.xml:47]: renamed "manifest"

File: input/resources/OperationDefinition-evaluate.xml [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+        <max value="1"/>
+        <documentation value="The end of the measurement period. The period will end at the end of the period implied by the supplied timestamp. E.g. a value of 2014 would set the period end to be 2014-12-31T23:59:59 inclusive"/>
+        <type value="date"/>
+    </parameter>
+    <parameter>
+        <name value="manifest"/>
+        <use value="in"/>
+        <min value="0"/>
+        <max value="1"/>
+        <targetProfile value="http://hl7.org/fhir/uv/crmi/StructureDefinition/crmi-manifestlibrary"/>
```
Comment (p9g) [inline input/resources/OperationDefinition-evaluate.xml:51]: target profile should be CQFMContentRelease which derives from CRMI manifest.  See https://jira.hl7.org/browse/FHIR-45309 not yet applied

File: input/resources/OperationDefinition-evaluate.xml [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+        <type value="id"/>
+    </parameter>
+    <parameter>
+        <name value="measureIdentifier"/>
+        <use value="in"/>
+        <min value="0"/>
+        <max value="*"/>
+        <documentation value="The business identifier of the measure to be evaluated. E.g., CMS125 is a CMS assigned identifier for the Breast Cancer Screening eCQM in their system."/>
+        <type value="string"/>
+        <searchType value="token"/>
```
Comment (p9g) [inline input/resources/OperationDefinition-evaluate.xml:82]: Where do we specify that this token is related to https://build.fhir.org/ig/HL7/cqf-measures/StructureDefinition-publishable-measure-cqfm-definitions.html#Measure.identifier:publisherIdentifier ?

**Comment by dand9959** [comment]
Thanks for all the comments. They are quite helpful.

We've encountered some technical issues on the forked branch that make the conflict resolution  in this PR  intractable.  After discussion, we have decided to create a new, clean PR to replace this one. The new PR will implement all of the changes reflected in this request, including additions per the comments here. The PR will be forthcoming on 08/06/2024

**Comment by dand9959** [comment]
**PLEASE BE ADVISED:**   
[Redo of PR136 fhir 40032 fhir 46287 by dand9959 · Pull Request #144 · HL7/davinci-deqm (github.com)](https://github.com/HL7/davinci-deqm/pull/144)  is the new (correct) PR to use instead of this one.

File: input/resources/OperationDefinition-evaluate.xml (hunk 3 above)
Comment (p9g) [reply input/resources/OperationDefinition-evaluate.xml:51]: This did not make it into PR144, perhaps because dependency is on QMIG ballot-2?

**Comment by p9g** [comment]
Since PR144 is merged, this PR136 should be closed without merge

**Comment by dand9959** [comment]
Closed without merge.  PR 144 was used instead.

**Comment by dand9959** [comment]
Done.
________________________________

**Pull request closed without merging**